<launch>
  <!-- ******************************************************************* -->
  <!-- ARGUMENTS (can be overridden from outside) -->

  <!-- Flatland -->
  <arg name="world_path" default="$(find pedestrian_controller)/config/conestogo_office_world.yaml"/>
  <arg name="update_rate" default="100.0"/>
  <arg name="viz_pub_rate" default="30.0"/>
  <arg name="step_size" default="0.01"/>
  <arg name="show_viz" default="true"/>
  <arg name="use_rviz" default="false"/>  

  <!-- Pedestrian controller -->
  <arg name="pedestrian_model_path" default="$(find pedestrian_controller)/config/models/own/turtlebot_model_clone.yaml"/>
  <arg name="pedestrian_count" default="10"/>

  <!-- Other -->
  <env name="ROSCONSOLE_FORMAT" value="[${severity} ${time} ${logger}]: ${message}" />
  <param name="use_sim_time" value="true"/>  

  <!-- ******************************************************************* -->
  <!-- NODES -->

  <!-- Flatland server -->
  <node name="flatland_server" pkg="flatland_server" type="flatland_server" output="screen">
    <param name="world_path" value="$(arg world_path)" />
    <param name="update_rate" value="$(arg update_rate)" />
    <param name="viz_pub_rate" value="$(arg viz_pub_rate)" />
    <param name="step_size" value="$(arg step_size)" />
    <param name="show_viz" value="$(arg show_viz)" />
  </node>

  <!-- Flatland visualization (if enabled) -->
  <group if="$(arg show_viz)">
    <node name="flatland_viz" pkg="flatland_viz" type="flatland_viz" output="screen" required="true" unless="$(arg use_rviz)"/>
  </group>

  <!-- Pedestrian controller (Python)-->
  <node name="pedestrian_controller" pkg="pedestrian_controller" type="pedestrian_controller.py" output="screen">
    <param name="pedestrian_model_path" value="$(arg pedestrian_model_path)" />
    <param name="pedestrian_count" value="$(arg pedestrian_count)" />
  </node>

  <!-- Spawn pedestrian with ros service -->
  <!-- <node name="spawn_model" pkg="rosservice" type="rosservice" args="call -\-wait /spawn_model &quot;{
    yaml_path: '$(find flatland_agent_spawner)/config/turtlebot_model.yaml',
    name: 'turtlebot1',
    ns: 'test',
    pose: {x: 0.0, y: 0.0, theta: 0.0}}&quot;" /> -->
</launch>
